### PR TITLE
[ticket/14403] Don't expect user_id and user_ip in phpbb\log

### PR DESCRIPTION
**Pull request by Elsensee**
[PHPBB3-14403](https://tracker.phpbb.com/browse/PHPBB3-14403)

### CODE REVIEW

**Comment by rxu** [comment]
This don't actually solve the issue, as accessing the ```$user->data['user_id']``` which is not set in outer script (like cron called via CLI) will throw error itself.
Thus an outer script will require the same workaround with id/ip still.

**Comment by Elsensee** [comment]
Indeed, but E_NOTICE won't exit the script.

**Comment by Nicofuma** [comment]
For that we need to fake the $user values when we are using the CLI. It can be done using the event ConsoleEvents::COMMAND

**Comment by marc1706** [comment]
Why do you want to fake that with an event? The user object already exists in the cli and a session is not being created. I don't see a reason why we should not just inject the data there.

**Comment by Nicofuma** [comment]
I want to fake the user data (data['user_ip'], data['user_id'] and eventually other data that are not present in the CLI context because of the lack of session) because otherwise we will throw notices (or other errors later) in many places (it is already the case actually). And also because I think it's better to have a global workaround for this kind of issue instead of having a different workaround everywhere.

**Comment by Elsensee** [comment]
Marc proposed the same thing actually but he wanted to change these variables in bin/phpbbcli.php already, if I got him correct.

**Comment by Nicofuma** [comment]
Yep but I don't like putting more stuff in phpbbcli and the other frontends (especially because I'm planning to nuke a lot of things (like common) and wrap everything with a single "kernel" object (but that's for 3.3 so we have time to see ;)))

**Comment by marc1706** [comment]
Using an event for that just seems like a huge workaround for something that would just require 2 lines of code in the cli. Also, you'd have to add that event, too. IMHO there is no point for now in using an event for that really.

**Comment by Nicofuma** [comment]
(just to say, the event already exists. It's one of Symfony's dedicated to the CLI workflow)

**Comment by Elsensee** [comment]
So how do you want me to fake it, then? :D

**Comment by Nicofuma** [comment]
just set the data in the user object :)

**Comment by Elsensee** [comment]
Oh, err, btw: Updated. ;)

File: phpBB/bin/phpbbcli.php [hunk 1]
```diff
@@ -59,6 +59,7 @@
 require($phpbb_root_path . 'includes/compatibility_globals.' . $phpEx);
 
 $user = $phpbb_container->get('user');
+$user->data['user_id'] = ANONYMOUS;
 $user->add_lang('acp/common');
```
Comment (marc1706) [inline phpBB/bin/phpbbcli.php]: Please also add ```$user->ip = '127.0.0.1';```

Comment (Elsensee) [inline phpBB/bin/phpbbcli.php]: Ah, I saw that $user->ip was already set to an empty string and thought "no work necessary here" but I guess that makes more sense. :D

**Pull request merged**